src/components/Modal/Modal.test.tsx

```diff
@@ -9,8 +9,11 @@ import '../../../jest/helpers/removeModalTransitionStylesJestSerializer';
 
 const { DefaultInteractive } = composeStories(stories);
 
-// Required because the modal uses react portal under the hood.
-require('intersection-observer');
+window.ResizeObserver = class FakeResizeObserver {
+  observe() {}
+  unobserve() {}
+  disconnect() {}
+};
```

Comment:
How headless tests as well https://github.com/tailwindlabs/headlessui/commit/9ecd8dd9261e3f474b3961f2a2f03d43c3197180#diff-7cd642ada4942212b7b7b991e9f456e7c522e21afe805fedefa4966acf5d1ae4R47

Reply:
- We could probably run this thru Jest, which would set up any spies and such in case we wanted to assert any calls to the functions ([see here](https://jestjs.io/docs/es6-class-mocks)
- Usually when overriding objects on `window` it's good to use the set/reset pattern, where we place the previous value in a temp variable, and use `beforeEach`/`afterEach` to restore

```js

const origResizeObserver = window.ResizeObserver;
window.ResizeObserver = class Fake {...}

// ... and later reset the value back to originalResizeObserver
```

Reply:
- I didn't want to set spies as ResizeObserver is being used in Headless internally, and would test Headless functionality rather than behavior.
- Thought tear down wouldn't be necessary since a new JSDom is created for every test file and felt weird reassigning `window.ResizeObserver` to `undefined`